src/irmin-pack/unix/file_manager.ml

```diff
@@ -761,4 +779,41 @@ struct
     | T15 ->
         (* Unreachable *)
         assert false
+
+  (* TODO : this does not work yet with GC. *)
+  let split t =
```

Comment:
I think we ought to protect against call `split` when the appendable chunk is 0 length (ie disallow creating multiple 0 length chunks). We could enforce it here or in `Chunked_suffix`, just depends on where in the stack we think the invariant should exist. Perhaps a new error? WDYT?